avr-hal-generic/src/eeprom.rs

```diff
@@ -0,0 +1,339 @@
+//!HAL abstraction for EEPROM
+//!
+use core::marker;
+
+#[derive(ufmt::derive::uDebug, Debug)]
+pub struct OutOfBoundsError;
+
+/// Internal trait for low-level EEPROM peripherals.
+///
+/// This trait defines the common interface for all EEPROM peripheral variants.
+pub trait EepromOps<H> {
+    const CAPACITY: u16;
+
+    /// **Warning**: This is a low-level method and should not be called directly from user code.
+    fn raw_read_byte(&self, address: u16) -> u8;
+    /// **Warning**: This is a low-level method and should not be called directly from user code.
+    fn raw_write_byte(&mut self, address: u16, data: u8);
+    /// **Warning**: This is a low-level method and should not be called directly from user code.
+    fn raw_erase_byte(&mut self, address: u16);
+}
+
+pub struct Eeprom<H, EEPROM> {
+    p: EEPROM,
+    _h: marker::PhantomData<H>,
+}
+
+impl<H, EEPROM> Eeprom<H, EEPROM>
+where
+    EEPROM: EepromOps<H>,
+{
```

Comment:
Okay, we can actually expose the capacity as a constant as well.  Just add the following inside the impl block:

```rust
impl<H, EEPROM> Eeprom<H, EEPROM>
where
    EEPROM: EepromOps<H>,
{
    pub const CAPACITY: u16 = EEPROM::CAPACITY;

    // ...
}
```

Reply:
review fixed